rfcs/rfc-using-global-store-for-form-components.md

```diff
@@ -0,0 +1,317 @@
+   * */
+  deleteItemFromCheckBoxList (context, payload: StoreAttributeModifyIF) {
+    let currentItems = context.getters.getNestedAttribute({ path: payload.path })
+    if (currentItems && currentItems.includes(payload.id)) {
+      const indexOfItem = currentItems.indexOf(payload.id)
+      currentItems.splice(indexOfItem, 1)
+      const newPayload = {
+        id: null,
+        path: payload.path,
+        value: currentItems
+      }
+      context.commit('setNestedAttribute', newPayload)
+    }
+  }
+```
+
+Since the location to the store could be deeply nested, recursive functions are used to get and set the
+requested attribute.  These functions should be modified to receive the store location in dot notation
+rather than arrays. For example, this, `location.in.the.store` instead of this, `["location", "in", "the", "store"]`.
+
+#### Nested functions
+```javascript
+  /** recursive function that gets a nested store attribute when
+   * passed the location in the store represented as an array of
+   * strings like this:
+   * getProp(store, ["location", "in", "the", "store", "attribute"])
+   * The requested attribute can be nested multiple levels deep.
+   * */
+  getProp (store, nestedProperty) {
+    const prop = nestedProperty.shift()
+    if (!store[prop] || !nestedProperty.length) {
+      return store[prop]
+    }
+    return this.getProp(store[prop], nestedProperty)
+  }
+
+  /** recursive function that sets a nested store attribute when
+   * passed the location in the store represented as an array of
+   * plus the new value to be saved. Like this:
+   * setProp(store, ["location", "in", "the", "store", "attribute"], new_value)
+   * The requested attribute can be nested multiple levels deep.
+   * */
+  setProp (store, nestedProperty, value) {
+    const prop = nestedProperty.shift()
+    if (!store[prop]) {
+      Vue.set(store, prop, {})
+    }
+    if (!nestedProperty.length) {
+      if (value && typeof value === 'object' && !Array.isArray(value)) {
+        store[prop] = { ...store[prop], ...value }
+      } else {
+        store[prop] = value
+      }
+      return
+    }
+    this.setProp(store[prop], nestedProperty, value)
+  }
+```
+
+We envision expanding the low-level components like `SingleCheckbox.vue` to include component
+to display radio buttons, text fields and date fields etc.  Each of these components would talk
+directly to the store using the store getter `getNestedAttribute` and the mutator `setNestedAttribute`.
+
+The new approach has more advantages as the forms get more complex and have more nested objects.  For
+example:
+
+```html
+<template>
+  <FormRow>
+    <InLineCheckBox id="corp_owner" :path="path" :option="true">
+        Owned by corporate entity
+    </InLineCheckBox>
+  </FormRow>
+  <!-- corporation sub-component start -->
+  <FormRow v-if="getNestedAttribute(path, 'corp_owner') === true">
```

Comment:
I do like this, but I think going the composable route is a lot simpler. For example, you would just import the 'path' composable and check the 'corp_owner' global (or local to the parent flow if possible) inside it. It also means the globals you have aren't all defined in one giant store that everything is accessing constantly (I'm picturing looking in there and getting overwhelmed haha)